src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationPerformanceMonitor.cs

```diff
@@ -186,5 +295,42 @@ public void Dispose()
                 }
             }
         }
+
+        private class MemoryDelta : IDisposable
+        {
+            private long startingAllocatedMemory;
+            private long startingReservedMemory;
+            private long startingUnusedMemory;
+
+            private MemoryUsage memoryUsage = null;
+
+            public MemoryDelta(MemoryUsage memoryUsage)
+            {
+                if (!Profiler.enabled)
+                {
+                    UnityEngine.Debug.LogError($"Profiler not enabled, MemoryUsage not supported.");
+                    return;
+                }
+
+                this.memoryUsage = memoryUsage;
+                startingAllocatedMemory = Profiler.GetTotalAllocatedMemoryLong();
+                startingReservedMemory = Profiler.GetTotalReservedMemoryLong();
+                startingUnusedMemory = Profiler.GetTotalUnusedReservedMemoryLong();
+            }
+
+            public void Dispose()
+            {
+                if (!Profiler.enabled ||
+                    memoryUsage == null)
+                {
+                    UnityEngine.Debug.LogError($"Profiler not enabled or memoryUsage was null, MemoryUsage not in usable state.");
+                    return;
+                }
+
+                memoryUsage.TotalAllocatedMemoryDelta += Profiler.GetTotalAllocatedMemoryLong() - startingAllocatedMemory;
+                memoryUsage.TotalReservedMemoryDelta += Profiler.GetTotalReservedMemoryLong() - startingReservedMemory;
+                memoryUsage.TotalUnusedReservedMemoryDelta += Profiler.GetTotalUnusedReservedMemoryLong() - startingUnusedMemory;
+            }

```

Comment:
The IDisposable pattern rules state that it should be safe to call Dispose multiple times without adverse side-effects. I think this should have a guard against that to prevent double-incrementing the memory allocation.  There's already a DisposableBase that correctly implements IDisposable in the SpatialAlignment part of SpectatorView, perhaps it should be moved to a place where this class can derive from it.

Reply:
DisposableBase is a class while this is a struct. I'll add a flag for now to check that the calculation isn't reperformed.